doc/python/shapes.md

```diff
@@ -716,5 +716,89 @@ fig.update_layout(
+
+You can create layout shapes programatically, but you can also draw shapes manually by setting the `dragmode` to one of the shape-drawing modes: `'drawline'`,`'drawopenpath'`, `'drawclosedpath'`, `'drawcircle'`, or `'drawrect'`. If you need to switch between different shape-drawing or other dragmodes (panning, selecting, etc.), modebar buttons can be added in the `config` to select the dragmode. If you switch to a different dragmode such as pan or zoom, you will need to select the drawing tool in the modebar to go back to shape drawing.
+
+This shape-drawing feature is particularly interesting for annotating graphs, in particular [image traces](/python/imshow) or [layout images](/python/images).
+
+Once you have drawn shapes, you can select and modify an existing shape by clicking on its boundary (note the arrow pointer). Its fillcolor turns to pink to highlight the activated shape and then you can 
+- drag and resize it for lines, rectangles and circles/ellipses
+- drag and move individual vertices for closed paths
+- move individual vertices for open paths.
+
+An activated shape is deleted by cliking on the `eraseshape` button.
+
+Drawing or modifying a shape triggers a `relayout` event, which [can be captured by a callback inside a Dash application](https://dash.plotly.com/interactive-graphing). 
+
+```python
+import plotly.graph_objects as go
+fig = go.Figure()
+text="Click and drag here <br> to draw a rectangle <br><br> or select another shape <br>in the modebar"
+fig.add_annotation(
+            x=0.5,
+            y=0.5,
+            text=text,
+            xref="paper",
+            yref="paper",
+            showarrow=False,
+            font_size=20
+)
+# shape defined programatically
+fig.add_shape(editable=True,
+              x0=-1, x1=0, y0=2, y1=3,
+              xref='x1', yref='y1')
+# define dragmode and add modebar buttons
+fig.update_layout(dragmode='drawrect')
+fig.show(config={'modeBarButtonsToAdd':['drawline',
+                                        'drawopenpath',
+                                        'drawclosedpath',
+                                        'drawcircle',
+                                        'drawrect',
+                                        'eraseshape'
+                                       ]})
+```
+
+### Style of user-drawn shapes
+
+The layout `newshape` attribute controls the visual appearance of new shapes drawn by the user. `newshape` attributes have the same names as layout shapes.
+
+```python
+import plotly.graph_objects as go
+fig = go.Figure()
+text="Click and drag<br> to draw a rectangle <br><br> or select another shape <br>in the modebar"
+fig.add_annotation(
+            x=0.5,
+            y=0.5,
+            text=text,
+            xref="paper",
+            yref="paper",
+            showarrow=False,
+            font_size=20
+)
+# shape defined programatically
+fig.add_shape(line_color='yellow',
+              fillcolor='turquoise',
+              opacity=0.8,
```

Comment:
You may consider using `opacity <= 0.5` here & below if you want to allow users to start drawing new shapes from inside of another shapes.

Reply:
thanks !